// nuxt-auth/dist/index.js
'use strict';

var _require = require('path');
var resolve = _require.resolve;

var _require2 = require('lodash');
var merge = _require2.merge;

function module$1(moduleOptions) {
  // Apply defaults
  var defaults = {
    login: {
      endpoint: 'auth/login',
      propertyName: 'token',
      session: false
    },
    logout: {
      endpoint: 'auth/logout',
      method: 'GET',
      paramTokenName: '',
      appendToken: false
    },
    user: {
      endpoint: 'auth/user',
      propertyName: 'user',
      paramTokenName: '',
      appendToken: false
    },
    storageTokenName: 'nuxt-auth-token',
    tokenType: 'Bearer',
    notLoggedInRedirectTo: '/login',
    loggedInRedirectTo: '/'
  };

  var options = merge(defaults, moduleOptions, this.options.auth);

  // Plugin
  this.addPlugin({ src: resolve(__dirname, '../templates/auth.plugin.js'), fileName: 'auth.plugin.js' });

  // Middleware
  this.addTemplate({ src: resolve(__dirname, '../templates/auth.middleware.js'), fileName: 'auth.middleware.js', options: options });

  // Store
  this.addTemplate({ src: resolve(__dirname, '../templates/auth.store.js'), fileName: 'auth.store.js', options: options });
}

module.exports = module$1;
